src/lib.rs

```diff
@@ -12,6 +12,14 @@
 //! This is very similar to how `futures::stream::unfold` creates a `Stream`
 //! implementation from a seed value and a future-returning closure.
 //!
+//! # Error behaviour
+//!
+//! If any of the [`Sink`] methods produces an error, the sink transitions to
```

Comment:
```suggestion
//! If any of the [`Sink`] methods produce an error, the sink transitions to
```

`methods` is plural, thus I would write `produce`, what do you think?